src/reflection.ts

```diff
@@ -760,12 +754,24 @@ export class Parser {
     // This will deserialize the vector of discriminator values
     const readDiscriminators = this.readVectorOfScalarsLambda(discriminatorField);
 
-    // This will read the vector of tables from the actual field
-    // How to deserialize each table will depend on the discriminator value at the same vector
-    // index position in the discriminator values field
-    const vectorLambda = this.readVectorOfTablesLambda(field);
+    // enum value -> specific type information
+    const specificTypes = new Map<
+      number,
+      { typeIndex: number; deserializer: (table: Table) => Record<string, any> | undefined }
+    >();
 
-    const unionDeserializers = this.createUnionDeserializers(fieldType, readDefaults);
+    for (const [enumVal, type] of this.getUnionTypes(fieldType)) {
+      const specificTypeIndex = type.index();
+      if (specificTypeIndex < 0) {
```

Comment:
This happens for the _none_ enum since it has no specific type. Instead of having these checks here, I've added a comment to this logic in `getUnionTypes` where we ignore the none value.